spec/hash-table.md

```diff
@@ -17,13 +31,23 @@ The hash circuit supports two modes:
 
 ### MPT Mode
 
-Compute the digest of two message words, as in Merkle trees. This type of entries is indentified by the control flag value of 0.
+Compute the digest of two message words, as in Merkle trees. This type of entries is identified by the control flag value of 0.
+
+**Initial capacity:** `0`.
+
+**Message encoding:** each message word is a field element; that is, each word is a digest from the previous MPT layer.
 
 ### Var-Len Mode
 
 Compute the digest of a variable-length message. One such entry is composed of consecutive rows with the same digest value, and where the control flag is not 0.
 
-The message is chunked into `W` words of `STEP/2` bytes packed into field elements. The words are given two-by-two on consecutive rows in the table, absorbing `STEP` bytes per row. The control flag on a given row indicates the number of message bytes remaining in the current and following rows. On the last row, `control <= STEP`.
+**Initial capacity:** `L * 2^64`, with `L` the message length in bytes (before padding).
```

Comment:
Note: as of this commit, the code uses `L` without `2^64`.

Reply:
Great! Many thanks